packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -724,3 +728,181 @@ describeNoCompat("SingleCommit Summaries Tests", (getTestObjectProvider) => {
+		const summaryAsObj = JSON.parse((compressorSummary as any).content);
+		assert(
+			summaryAsObj.sessions.length === 0,
+			"Shouldn't have any local sessions as all ids are unack'd",
+		);
+		assert(
+			summaryAsObj.clusters.length === 0,
+			"Shouldn't have any local clusters as all ids are unack'd",
+		);
+	});
+
+	it("Includes ack'd ids in summary", async () => {
+		const { mainContainer } = await createMainContainerAndSummarizer(provider, config);
+
+		const defaultDataStore = await requestFluidObject<ITestDataObject>(
+			mainContainer,
+			defaultDataStoreId,
+		);
+
+		const idCompressor: IIdCompressor = (defaultDataStore._root as any).runtime.idCompressor;
+		assert(idCompressor !== undefined, "IdCompressor should be present");
+		idCompressor.generateCompressedId();
+		defaultDataStore._root.set("key", "value");
+		const containerRuntime = defaultDataStore._context.containerRuntime as ContainerRuntime;
+		await provider.ensureSynchronized();
+
+		const { summary } = await containerRuntime.summarize({
+			runGC: false,
+			fullTree: false,
+			trackState: false,
+			summaryLogger: new TelemetryNullLogger(),
+		});
+
+		const compressorSummary = summary.tree[".idCompressor"];
+		assert(compressorSummary !== undefined, "IdCompressor should be present in summary");
+		const summaryAsObj = JSON.parse((compressorSummary as any).content);
+		assert(
+			summaryAsObj.sessions.length === 1,
+			"Should have a local session as all ids are ack'd",
+		);
+		assert(
+			summaryAsObj.clusters.length === 1,
+			"Should have a local cluster as all ids are ack'd",
+		);
+	});
+
+	it("Newly connected container synchronizes from summary", async () => {
+		const { mainContainer } = await createMainContainerAndSummarizer(provider, config);
+
+		const defaultDataStore = await requestFluidObject<ITestDataObject>(
+			mainContainer,
+			defaultDataStoreId,
+		);
+
+		const idCompressor: IIdCompressor = (defaultDataStore._root as any).runtime.idCompressor;
+		assert(idCompressor !== undefined, "IdCompressor should be present");
+		idCompressor.generateCompressedId();
+		defaultDataStore._root.set("key", "value");
+		const containerRuntime = defaultDataStore._context.containerRuntime as ContainerRuntime;
+		await provider.ensureSynchronized();
+
+		const { summary } = await containerRuntime.summarize({
+			runGC: false,
+			fullTree: false,
+			trackState: false,
+			summaryLogger: new TelemetryNullLogger(),
+		});
+
+		const compressorSummary = summary.tree[".idCompressor"];
+		assert(compressorSummary !== undefined, "IdCompressor should be present in summary");
+		const summaryAsObj = JSON.parse((compressorSummary as any).content);
+		assert(
+			summaryAsObj.sessions.length === 1,
+			"Should have a local session as all ids are ack'd",
+		);
+		assert(
+			summaryAsObj.clusters.length === 1,
+			"Should have a local cluster as all ids are ack'd",
+		);
+
+		const otherContainer = await provider.loadTestContainer(config);
```

Comment:
This container does not load from the summary generated by the first container. Calling `containerRuntime.summarize` will not submit the summary to the server so a container cannot load from this summary.
Take a look at this test that loads from the previous submitted summary's version - https://github.com/microsoft/FluidFramework/blob/a80b764bf1731f9468501f7829d8efee5e16a8b3/packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts#L147

Reply:
I've rewritten the test. Let me know if it looks right!